llama_index/callbacks/open_inference_callback.py

```diff
@@ -208,18 +208,22 @@ def on_event_end(
             for node_with_score in payload[EventPayload.NODES]:
                 node = node_with_score.node
                 score = node_with_score.score
-                self._trace_data.query_data.node_ids.append(node.hash)
+                self._trace_data.query_data.node_ids.append(node.node_id)
                 self._trace_data.query_data.scores.append(score)
                 self._trace_data.node_datas.append(
                     NodeData(
-                        id=node.hash,
+                        id=node.node_id,
                         node_text=node.text,
                     )
                 )
         elif event_type is CBEventType.LLM:
-            self._trace_data.query_data.response_text = str(
-                payload.get(EventPayload.RESPONSE, "")
-            ) or str(payload.get(EventPayload.COMPLETION, ""))
+            if chat_response := payload.get(EventPayload.RESPONSE):
+                response_text = chat_response.message.content
+            elif completion_response := payload.get(EventPayload.COMPLETION):
+                response_text = completion_response.text
+            else:
+                response_text = ""
+            self._trace_data.query_data.response_text = response_text
```

Comment:
The role is being included as part of the response text when calling `str` on an instance of `ChatResponse` or `CompletionResponse` directly. So all of the response text is of the form "assistant: ..."